Tests/DatadogTests/Datadog/Core/Utils/JSONEncoderTests.swift

```diff
@@ -29,4 +29,26 @@ class JSONEncoderTests: XCTestCase {
             XCTAssertEqual(encodedURL.utf8String, #"{"value":"https:\/\/example.com\/foo"}"#)
         }
     }
+
+    func testWhenEncoding_thenKeysFollowLexicographicOrder() throws {
+        struct Foo: Codable {
+            var one = 1
+            var two = 1
+            var three = 1
+            var four = 1
+
+            enum CodingKeys: String, CodingKey {
+                case one = "aaaaaa"
```

Comment:
i think our use case is keys with `.`, no? if we add keys with `.` here, maybe we can catch potential issues in next iOS versions

Reply:
True 👌. Added one more variant: `aaa.aaa` to cover this.